src/prop_current/antenna_gainsV1_0.py

```diff
@@ -0,0 +1,150 @@
+'''
+Created on 6 juil. 2017
+
+@author: Red Technologies
+Gerard Garnier
+'''
+import vincenty
+from ehata_its_wf import *
+from cmath import isnan
+
+def GetCBSDAntennaGain(AZ, antennaAzimut, antennaBeamWidth):
+    """
+    Returns the CBSD antenna gain (R2-SGN-20).
+    Inputs:
+        AZ                        azimuth angle (deg : 0 - 359)
+        antennaAzimut             (deg : 0 - 359)
+        antennaBeamWidth          3-dB antenna beamwidth (deg : 0 - 359)
+    Output:
+        CBSD antenna gain
+    """
+    teta = antennaAzimut - AZ
+    return -min(12 * pow(teta/antennaBeamWidth,2), 20)
+
+def GetFSSAntennaGain(pfl, h_b__meter, h_m__meter, antennaLat, antennaLon, FSSLat, FSSLon, antennaElevation, alpha, w1, w2):
+    """
+    Returns the FSS Earth Station Antenna Gain (R2-SGN-21).
+    Inputs:
+        pfl                       Terrain profile line
+        h_b__meter                height of the CBSD, in meters
+        h_m__meter                height of the FSS, in meters
+        antennaElevation          the FSS earth station antenna elevation (deg : 0 - 359)        
+        alpha                     FSS earth station antenna azimuth (deg : 0 - 359)
+        w1, w2                    weights 
+    Output:
+        FSS Earth Station Antenna Gain
+    """
+    d, azstart, AZ = vincenty.dist_bear_vincenty(antennaLat, antennaLon, FSSLat, FSSLon)
+
+    theta = [0., 0.]
+    GetElevationAngle(pfl, h_m__meter, h_b__meter, theta)
+    
+    theta[0] = npy.degrees(theta[0])
+    if(theta[0] < 0) :
+        theta[0]+=360
+            
+    """
+    off-axis angle
+    """
+    ThetaAngle = (180 * (1 / math.cos( (math.cos(theta[0]) * math.cos(antennaElevation)\
+                                        * math.cos(alpha - AZ)) + (math.sin(theta[0]) * math.sin(antennaElevation)) )))/ math.pi
+        
+    Ggso, GgsoT = GetGgsoAndGgsoT(ThetaAngle)
+    
+    """
+    If the FSS earth station registration data includes values for w1 and w2, 
+    SAS shall use these values. Otherwise SAS shall assume w1=0 and w2=1.
+    """
+    if isnan(w1) or isnan(w2):
+        FSSGain = GgsoT
+    else:
+        FSSGain = w1*Ggso + w2*GgsoT
+    
+    return FSSGain
+
+def GetElevationAngle(pfl, h_m__meter, h_b__meter, theta):
+    
+    """
+    Returns elevation angles of the horizons from each terminal.
+    Inputs:
+        pfl          Terrain profile line
+        h_m__meter   height of the mobile, in meters
+        h_b__meter   height of the base station, in meters
+    Output:
+        theta : Elevation angles
+                 - theta[0] = mobile horizon distance, in radians
+                 - theta[1] = base station horizon distance, in radians
+    """
+    np = int(pfl[0])           #// number of points
+    xi = pfl[1]                #// step size of the profile points, in meter
+    d_meter = np * xi
+
+    h_gnd__meter = AverageTerrainHeight(pfl)
```

Comment:
Why not be explicit instead of referencing another module:
` h_gnd_meter = npy.mean(pfl[2:])`
Also please add on top the import
`import numpy as npy`